test/sciml_style.jl

```diff
@@ -1434,4 +1493,540 @@
+        """
+        # Good - LHS tuple should stay on one line, function args can break
+        expected = raw"""
+        discrete_modified, saved_in_cb = DiffEqBase.apply_discrete_callback!(integrator,
+                                                                             integrator.opts.callback.discrete_callbacks...
+                                                                             )
+        """
+        @test format_text(str, SciMLStyle(), yas_style_nesting = true) == expected
+        
+        # Test when source already has split LHS - it should be preserved
+        str_split = raw"""
+        discrete_modified,
+        saved_in_cb = DiffEqBase.apply_discrete_callback!(integrator,
+                                                          integrator.opts.callback.discrete_callbacks...)
+        """
+        # Should preserve the split since join_lines_based_on_source = true
+        expected_split = raw"""
+        discrete_modified,
+        saved_in_cb = DiffEqBase.apply_discrete_callback!(integrator,
+                                                          integrator.opts.callback.discrete_callbacks...
+                                                          )
+        """
+        @test format_text(str_split, SciMLStyle(), yas_style_nesting = true) == expected_split
+
+        # Test 3: Function definition with Union type arguments
+        str = raw"""
+        function compute_gradient_correction_matrix!(corr::Union{GradientCorrection,
+                                                                 BlendedGradientCorrection,
+                                                                 MixedKernelGradientCorrection},
+                                                     another_argument)
+        end
+        """
+        # Good (main) - function name and first arg should stay on same line
+        expected = raw"""
+        function compute_gradient_correction_matrix!(corr::Union{GradientCorrection,
+                                                                 BlendedGradientCorrection,
+                                                                 MixedKernelGradientCorrection},
+                                                     another_argument)
+        end
+        """
+        # Current behavior breaks after function name
+        expected_current = raw"""
+        function compute_gradient_correction_matrix!(
+                                                     corr::Union{GradientCorrection,
+                                                                 BlendedGradientCorrection,
+                                                                 MixedKernelGradientCorrection},
+                                                     another_argument)
+        end
+        """
+        @test format_text(str, SciMLStyle(), yas_style_nesting = true) == expected_current
```

Comment:
```suggestion
        @test format_text(str, SciMLStyle(), yas_style_nesting = true) == expected
```
Would this be too difficult to fix in this PR?